libnmstate/nm/connection.py

```diff
@@ -146,15 +158,21 @@ def safe_activate_async(self):
         cancellable = self._mainloop.new_cancellable()
 
         active_conn = get_device_active_connection(self.nmdevice)
+        nm_acsreason = nmclient.NM.ActiveConnectionStateReason
         if active_conn:
             ac = ActiveConnection(active_conn)
-            if ac.is_activating:
+            if (
+                ac.is_activating
+                or ac.reason == nm_acsreason.DEVICE_DISCONNECTED
```

Comment:
This is a bit confusing, it looks like if a device is disconnected, we will wait for it to to get activated. On the other hand, a disconnected device is not expected to have an active-connection at all.
Not suggesting anything, just pointing out the confusion.

Reply:
ActivateConnection still can be exists for a short time with state==Deactivated reason=DEVICE_DISCONNECTED.